### PR TITLE
Fixes 5.0final/v1

### PR DESCRIPTION
**Pull request by shivan1b**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
https://redmine.openinfosecfoundation.org/issues/2386
https://redmine.openinfosecfoundation.org/issues/2669
https://redmine.openinfosecfoundation.org/issues/1940

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -32,7 +32,7 @@ install-conf:
 
 install-rules:
 if HAVE_SURICATA_UPDATE
-	$(DESTDIR)$(bindir)/suricata-update \
+	LD_LIBRARY_PATH="$(DESTDIR)$(prefix)/lib" $(DESTDIR)$(bindir)/suricata-update \
```
Comment (victorjulien) [inline Makefile.am]: Wonder if this could be ```$(libdir)```.

File: src/source-af-packet.c [hunk 2]
```diff
@@ -2009,7 +2009,7 @@ int AFPIsFanoutSupported(void)
     close(fd);
 
     if (r < 0) {
-        SCLogPerf("fanout not supported by kernel: %s", strerror(errno));
+        SCLogError(SC_ERR_INVALID_VALUE, "fanout not supported by kernel: Kernel too old or cluster-id %d already in use.", cluster_id);
```
Comment (victorjulien) [inline src/source-af-packet.c]: can you break this line to stay within ~80 chars?

File: src/suricata.c [hunk 3]
```diff
@@ -2831,6 +2837,23 @@ static int PostConfLoadedSetup(SCInstance *suri)
...
+
+    if (ConfigCheckLogDirectoryExists(suri->log_dir) != TM_ECODE_OK) {
+        SCLogError(SC_ERR_LOGDIR_CONFIG, "The logging directory \"%s\" "
+                "supplied by %s (default-log-dir) doesn't exist. "
+                "Shutting down the engine", suri->log_dir, suri->conf_filename);
+        SCReturnInt(TM_ECODE_FAILED);
+    }
+    if (!IsLogDirectoryWritable(suri->log_dir)) {
+        SCLogError(SC_ERR_LOGDIR_CONFIG, "The logging directory \"%s\" "
+                "supplied by %s (defailt-log-dir) is not writable. "
```
Comment (victorjulien) [inline src/suricata.c]: typo: defailt

**Comment by shivan1b** [comment]
Replaced with #4280

**Pull request closed without merging**